### PR TITLE
Curves: Prepare serialization

### PR DESCRIPTION
**Pull request by Mugen87**
This PR prepares curve related entities for serialization/deserialization (needed for #8007). It adds the `.type` attribute analogous to other classes in `three.js`. 

There is one problematic point. `CatmullRomCurve3` already used `.type` for defining the type of a catmull–rom spline. I've introduced `.curveType` for this task and use `.type` for the class type definition now.

So this change needs to be noted in the migration guide.

### CODE REVIEW

File: examples/webgl_geometry_extrude_shapes.html [hunk 1]
```diff
@@ -74,7 +74,6 @@
 					new THREE.Vector3(  60, -100, -60 )
 				] );
 
-				closedSpline.type = 'catmullrom';
```
Comment (Mugen87) [inline examples/webgl_geometry_extrude_shapes.html]: `catmullrom` is now the default value of `.curveType`.

**Comment by WestLangley** [comment]
>I've introduced .curveType for this task and use .type for the class type definition now.

Perhaps adding a setter for `CatmullRomCurve3.type` to `Legacy.js` would be good idea.

**Comment by Mugen87** [comment]
How would this work? When i define a setter for `type`, i would need a an additional or private property in order to set the given value, right? I'm not sure if `_type` could be annoying in some way...
```javascript
Object.defineProperties( CatmullRomCurve3.prototype, {

	type: {
		get: function () {

			return this._type;

		},
		set: function ( value ) {

			if ( value === 'centripetal' || value === 'chordal' || value === 'catmullrom' ) {

				console.warn( 'THREE.CatmullRomCurve3: .type is now .curveType.' );
				this.curveType = value;

			} else {

				this._type = value;

			}

		}
	}

} );
```

**Comment by WestLangley** [comment]
Try something like this, instead:
```
Object.defineProperties( THREE.CatmullRomCurve3.prototype, {

	type: {

		set: function ( value ) {

			if ( value !== 'CatmullRomCurve3' ) {

				console.warn( 'THREE.CatmullRomCurve3: .type is now .curveType.' );
				this.curveType = value;

			}

		}
	}

} );
```

**Comment by Mugen87** [comment]
With this code the value of `type` will be always `undefined`. Besides, the warning logs also the value `Curve` from the `Curve` constructor, which is obviously wrong.

**Comment by Mugen87** [comment]
From https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Functions/set
_It is not possible to simultaneously have a setter on a property that holds an actual value._

**Comment by WestLangley** [comment]
Sorry, I should have thought about this more carefully before I posted... :/

**Comment by looeee** [comment]
Would it be possible to revive this PR? For consistency with the rest of the library curves should have copy and clone functionality.

**Comment by Mugen87** [comment]
I can create a new PR with this changes on the latest `dev` branch. But i don't want to invest that time if it does not get merged anyway. If @mrdoob is okay with the changes, i'll give it a go. 👍

**Comment by mrdoob** [comment]
Go for it 👍

**Comment by Mugen87** [comment]
Added new PR 👍

**Pull request closed without merging**